rfd/00xx-pod-access-requests.md

```diff
@@ -0,0 +1,555 @@
+
+Once the user has the certificate with the new group embedded, Teleport will
+forward the certificate details into the Kubernetes cluster as Impersonation Headers.
+
+When the access request expires, Teleport will clean the temporary Kubernetes
+RBAC `Role` and `RoleBinding` objects.
+
+## UX: Request Flow
+
+Extending [RFD 59](https://github.com/gravitational/teleport/blob/master/rfd/0059-search-based-access-requests.md),
+Teleport will consider Kubernetes Pods as native resources.
+This allows users to search Pods that normally they do not
+have access to and request access to them. Besides requesting access to
+specific Pods - using the Pod full name, Teleport allows users to request
+access to Pods that follow a pattern like Pods automatically created from a
+Deployment - `deploymentname-*-*`.
+
+```bash
+$ tsh request search --kind pod --search pod
+
+Found 3 items:
+name kind     id
+pod-1 pod     pod:pod-1
+dep-1 pod     node:dep-1
+dep-2 pod     node:dep-2
+
+$ tsh request create --resources "pod:pod1,pod:dep-*"
+Waiting for request to be approved...
+Approved!
+$ kubectl get pods 
+```
+
+Similarly to Resource Based Access Requests, Pod Access requests will leverage
+`search_as_roles` to control which Pods the user can view while searching.
+
+```yaml
+kind: role
+metadata:
+  name: response-team
+spec:
+  allow:
+    request:
+      # search_as_roles allows a member of the response team
+      # to search for resources accessible to users with the kube-admins role,
+      # which they will be allowed to request access to. This includes Pods in 
+      # the target Kubernetes Cluster.
+      search_as_roles: [kube-admin]
+```
+
+```yaml
+kind: role
+metadata:
+  name: kube-admin
+spec:
+  allow:
+    # kubernetes_groups will be used by Teleport to fetch the Pods available in 
+    # the target cluster. The kubernetes_groups field is used by Kubernetes RBAC 
+    # to limit the Pods the user can request access to.
+    kubernetes_groups: ["system:masters"]
+    # kubernetes_labels defines which Kubernetes clusters the user is able to 
+    # request Pod access into.
+    kubernetes_labels:
+       owner: prod_team
+```
+
+While requesting access to a Pod or searching for Pods, the user must identify
+which cluster he wants to request access to. Request access to multi-clusters
+will not be supported.
+
+For now, all Pod access requests come from `tsh`, but the functionality can be
+extended in the future using a `kubectl` plugin.
+
+## Option 1: Restrict Pod access
+
+Proposed is a new way to restrict access to specific Pods on a Kubernetes cluster.
+Given a set of roles with Kubernetes RBAC principals - groups or users,
+that normally grant access to a larger set of Pods, once `kubernetes_pods`
+is specified in at least one of the roles, Teleport will restrict the user's
+access to the list of pods specified under `kubernetes_pods`.
+`kubernetes_pods` must define a subset of the Pods allowed by the user's Kubernetes
```

Comment:
Nit: Technically it doesn't have to. Teleport will let the request through but Kubernetes reject it. I would just say that user's "effective" permissions will be an intersection of what Teleport RBAC and Kubernetes RBAC allow.